catchup/service.go

```diff
@@ -710,6 +728,15 @@ func (s *Service) fetchRound(cert agreement.Certificate, verifier *agreement.Asy
 				return
 			default:
 			}
+			if err == errNoBlockForRound {
+				// remote peer doesn't have the block, try another peer
+				// quit if the the same peer peer encountered errNoBlockForRound more than errNoBlockForRoundThreshold times
+				if count := peerErrors[peer]; count > errNoBlockForRoundThreshold {
+					s.log.Debugf("fetchAndWrite(%d): remote peers do not have the block. Quitting", cert.Round)
+					return
```

Comment:
Spoke live about this, thinking about introducing a 1 second sleep duration (rather than returning and falling back to the value of deadline timeout). Assumption is that block will likely be available *very soon*.

Reply:
Thanks, added and updated the PR description.